src/metabase/api/search.clj

```diff
@@ -215,6 +219,17 @@
       (h/merge-left-join [Collection :collection]
                          [:= collection-id-column :collection.id]))))
 
+(s/defn ^:private add-table-db-id-clause
+  "Add a WHERE clause to only return tables with the given DB id.
+  Used in data picker for joins because we can't join across DB's."
+  [query :- su/Map, id :- (s/maybe s/Int)]
+  (if (some? id) (h/merge-where query [:= id :db_id]) query))
+
+(s/defn ^:private add-card-db-id-clause
+  "Add a WHERE clause to only return crads with the given DB id.
+  Used in data picker for joins because we can't join across DB's."
+  [query :- su/Map, id :- (s/maybe s/Int)]
+  (if (some? id) (h/merge-where query [:= id :database_id]) query))
```

Comment:
(deleted an incorrect comment since I misread this)

(Since I misread it, maybe it's worth breaking the `if`s into three lines instead of one)